src/Reflection/ReflectionObject.php

```diff
@@ -64,10 +65,17 @@ public static function createFromInstance($object) : ReflectionClass
         $className = get_class($object);
 
         if (strpos($className, ReflectionClass::ANONYMOUS_CLASS_NAME_PREFIX) === 0) {
-            $reflector = new ClassReflector(new AnonymousClassObjectSourceLocator(
-                $object,
-                (new BetterReflection())->phpParser()
-            ));
+            $reflector = new ClassReflector(
+                new AggregateSourceLocator(
+                    [
+                        (new BetterReflection())->sourceLocator(),
```

Comment:
Note: the problem with this diff is that we have multiple `(new BetterReflection())` instances here, which means that these source locators are operating on disjunct sources.

Reply:
Te better clarify scope, use a `StringSourceLocator` and reproduce the problem within a single string containing an anonymous class: you will see a crash (can't find the parent class)